pysr/ssr.py

```diff
@@ -0,0 +1,171 @@
+            )
+        if isinstance(weights, np.ndarray) and len(weights) != len(X):
+            raise ValueError("The length of `weights` must have shape (n_times,).")
+        y = X.copy()
+        temp = X.copy()[0]
+        X = np.lib.stride_tricks.sliding_window_view(
+            y[:-1].flatten(), self.recursive_history_length * temp.shape[0]
+        )[:: temp.shape[0], :]
+        y = np.array([i for i in y[self.recursive_history_length :]])
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+
+        if not variable_names:
+            if len(temp.shape) == 0:
+                variable_names = [
+                    f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)
+                ]
+            elif len(temp.shape) == 1:
+                variable_names = [
+                    f"x{i}t_{j}"
+                    for j in range(temp.shape[0])
+                    for i in range(self.recursive_history_length, 0, -1)
+                ]
+        else:
+            variable_names = [
+                i + "t_" + str(j)
+                for i in variable_names
+                for j in range(self.recursive_history_length, 0, -1)
+            ]
+        super().fit(
+            X,
+            y,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+
+        return self
+
+    def predict(self, X, index=None):
+        """
+        Predict y from input X using the equation chosen by `model_selection`.
+
+        You may see what equation is used by printing this object. X should
+        have the same columns as the training data.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training data of shape `(n_times, 1)`.
+        index : int | list[int]
+            If you want to compute the output of an expression using a
+            particular row of `self.equations_`, you may specify the index here.
+            For multiple output equations, you must pass a list of indices
+            in the same order.
+
+        Returns
+        -------
+        x_predicted : ndarray of shape (n_samples, nout_)
+            Values predicted by substituting `X` into the fitted sequence symbolic
+            regression model.
+
+        Raises
+        ------
+        ValueError
+            Raises if the `best_equation` cannot be evaluated.
+        """
+        if len(X) < self.recursive_history_length:
+            raise ValueError(
+                f"Recursive symbolic regression with a history length of {self.recursive_history_length} requires at least {self.recursive_history_length} datapoints."
+            )
+        temp = X.copy()
+        X = np.lib.stride_tricks.sliding_window_view(
```

Comment:
Currently in `requirements.txt`, we have the constraint that

```
numpy>=1.13.0,<3.0.0
```

However, it looks like this function was only introduced in numpy 1.20.0: https://numpy.org/doc/stable/reference/generated/numpy.lib.stride_tricks.sliding_window_view.html. So could you also update the requirements to limit numpy to >=1.20.0?